tutormfe/templates/mfe/build/mfe/env.config.jsx

```diff
@@ -0,0 +1,51 @@
+{{- patch("mfe-env-config-buildtime-imports") }}
+
+function addPlugins(config, slot_name, plugins) {
+  if (slot_name in config.pluginSlots === false) {
+    config.pluginSlots[slot_name] = {
+      keepDefault: true,
+      plugins: []
+    };
+  }
+
+  config.pluginSlots[slot_name].plugins.push(...plugins);
+}
+
+{{- patch("mfe-env-config-buildtime-definitions") }}
+
+async function getConfig () {
+  let config = {
+    pluginSlots: {}
+  };
+
+  try {
+    /* We can't assume FPF exists, as it's not declared as a dependency in all
+     * MFEs, so we import it dynamically. In addition, for dynamic imports to
+     * work with Webpack all of the code that actually uses the imported module
+     * needs to be inside the `try{}` block.
+     */
+    const { DIRECT_PLUGIN, PLUGIN_OPERATIONS } = await import('@openedx/frontend-plugin-framework');
+
+    {{- patch("mfe-env-config-runtime-definitions") }}
+
+    {%- for slot_name, plugin_config in iter_plugin_slots("all") %}
+    addPlugins(config, '{{ slot_name }}', [{{ plugin_config }}]);
```

Comment:
Is there any particular reason for adding plugin_config object one by one?

Could we pass array for one plugin_slot (keeping in mind multiple plugins can have same slots)?
<img width="786" alt="image" src="https://github.com/user-attachments/assets/18f8cff5-5bc3-4a25-8087-df797c5a80f3">

Like this:
```
PLUGIN_SLOTS.add_items([
        (
            mfe,
            "footer_slot",
            """
            [
    {
        op: PLUGIN_OPERATIONS.Hide,
        widgetId: 'default_contents',
    }
    {
      op: PLUGIN_OPERATIONS.Insert,
      widget: {
        id: 'default_contents',
        type: DIRECT_PLUGIN,
        priority: 1,
        RenderWidget: <Footer />,
      },
    },
    {
      op: PLUGIN_OPERATIONS.Insert,
      widget: {
        id: 'read_theme_cookie',
        type: DIRECT_PLUGIN,
        priority: 2,
        RenderWidget: AddDarkTheme,
      },
    },
  ]"""
        ),
    ])
```

Reply:
Apologies, I just realized that objects are already passing in array. I need to pass without array brackets:
```javascript

PLUGIN_SLOTS.add_items([
        (
            mfe,
            "footer_slot",
            """
            
    {
        op: PLUGIN_OPERATIONS.Hide,
        widgetId: 'default_contents',
    },
    {
      op: PLUGIN_OPERATIONS.Insert,
      widget: {
        id: 'default_contents',
        type: DIRECT_PLUGIN,
        priority: 1,
        RenderWidget: <Footer />,
      },
    },
    {
      op: PLUGIN_OPERATIONS.Insert,
      widget: {
        id: 'read_theme_cookie',
        type: DIRECT_PLUGIN,
        priority: 2,
        RenderWidget: AddDarkTheme,
      },
    },
  """
        ),
    ])
```

Reply:
There are different ways you can load the configuration in.  @regisb suggested we document it one-by-one to avoid issues with trailing commas.  Though in this latest iteration of `env.config.jsx` it doesn't really matter if there are trailing commas or not, I guess it keeps it simple.

Otherwise, yes you can totally use a single `add_item()` with a string that has multiple ops for one slot.